tests/oc_template_test.go

```diff
@@ -0,0 +1,328 @@
+			},
+			{
+				"name": "MEMORY",
+				"value": "4096Mi",
+				"description": "Amount of memory"
+			},
+			{
+				"name": "CPU_CORES",
+				"value": "4",
+				"description": "Amount of cores"
+			}
+		],
+		"metadata": {
+			"labels": {
+				"miq.github.io/kubevirt-is-vm-template": "true",
+				"kubevirt.io/os": "fedora27"
+			},
+			"annotations": {
+				"description": "OCP KubeVirt Fedora 27 VM template",
+				"tags": "kubevirt,ocp,template,linux"
+			},
+			"console.openshift.io/iconClass": "icon-fedora",
+			"name": "fedora-vm-template"
+		}
+	}`
+
+	Context("with oc command", func() {
+		var vmTemplateJsonFile = ""
+		var vmJsonFile = ""
+
+		BeforeEach(func() {
+			tests.SkipIfNoOc()
+			// write testing vm template into file
+			vmTemplateJsonFile, err = WriteJson("testvm-template", vmTemplate)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			if vmTemplateJsonFile != "" {
+				// remove testing vm template
+				err = os.Remove(vmTemplateJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmTemplateJsonFile = ""
+			}
+			if vmJsonFile != "" {
+				// remove testing vm JSON file
+				err = os.Remove(vmJsonFile)
+				Expect(err).ToNot(HaveOccurred())
+				vmJsonFile = ""
+			}
+		})
+
+		It("should convert an offline vm template into JSON file", func() {
+			By("Converting offline vm template into JSON file via oc-process command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			vmJsonFile, err = WriteJson(vmName, out)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should create an offline vm from a template", func() {
+			var message = ""
+			// probably there is a better way how to do this without duplicating the code
+			By("Converting offline vm template into JSON file via oc-process command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			vmJsonFile, err = WriteJson(vmName, out)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating offline vm from JSON file via oc-create command")
+			out, err = tests.RunOcCommand("process", "-f", vmTemplateJsonFile, vmTemplateParams)
+			Expect(err).ToNot(HaveOccurred())
+			message = fmt.Sprintf("offlinevirtualmachine \"%s\" created", vmName)
+			Expect(out).To(Equal(message))
+
+			// this should be probably handled differently
+			time.Sleep(time.Second * 120)
+
+			// teardown
+			By("Deleting the vm via oc-delete command")
+			out, err = tests.RunOcCommand("delete", "-f", vmJsonFile)
+			Expect(err).ToNot(HaveOccurred())
+			message = fmt.Sprintf("offlinevirtualmachine \"%s\" deleted", vmName)
+			Expect(out).To(Equal(message))
+
+			// this should be probably handled differently
+			time.Sleep(time.Second * 120)
+
+			By("Checking out that the vm does not exist anymore")
+			result := virtClient.RestClient().Get().Resource(tests.VmResource).Namespace(k8sv1.NamespaceDefault).Name(vmName).Do()
```

Comment:
Could you use one of our test namespaces for testing the entitis? Normally you create everything in `tests.NamespaceTestDefault`.